src/api.c

```diff
@@ -886,6 +889,23 @@ struct nmd_frame *nmd_get_next_frame(struct nmd_ctx *s)
         return ret_frame(s, NULL);
 
     AVFrame *frame = pop_frame(s);
+    if (!frame && s->eof) {
+        /*
+         * If the async module reached EOF and there is no more frame to
+         * return, we need to seek back to the beginning of the file because
+         * nmd_get_next_frame() is expected to loop around when reaching EOF.
+         */
+        s->eof = 0;
+
```

Comment:
So this is going to cause a (costly) seek systematically at the end. Shouldn't that happen only if get_next_frame is called again after a `NULL` was returned?

Reply:
It depends on your scenario, sending an seek command asynchronously at the end will make the next get_next_frame() faster.

Reply:
Right but the classic scenario of `get_next_frame()` is to go through all the frame and stop at the end. This scenario get hurt a lot and the user has no way of avoiding it. And in the scenario where the user wants the loop, the seek must be done anyway, so I think that's fine to have it lazily.

Reply:
Done.